QSB/TransformSync/TransformSync.cs

```diff
@@ -80,18 +82,17 @@ private void Update()
                 return;
             }
 
-            // Get which sector should be used as a reference point
-            var sectorTransform = PlayerRegistry.LocalPlayer.ReferenceSector;
-
-            if (sectorTransform == null)
-            {
-                DebugLog.ToConsole($"Error - Player ID {PlayerRegistry.LocalPlayer.NetId}'s reference sector is null!", MessageType.Error);
-            }
+            var sectorTransform = PlayerRegistry.GetPlayer(GetAttachedNetId()).ReferenceSector;
 
             if (hasAuthority) // If this script is attached to the client's own body on the client's side.
             {
                 transform.position = sectorTransform.InverseTransformPoint(SyncedTransform.position);
                 transform.rotation = sectorTransform.InverseTransformRotation(SyncedTransform.rotation);
+
+                if (Override())
```

Comment:
I suggest you make Override into a virtual and empty Method that only the probe overrides, and that's called from here. Also rename it to something else.

Reply:
That way you don't have to make the other (not probe) TransformSyncs uglier